testing/end-to-end/tests/playos-update-process.nix

```diff
@@ -0,0 +1,264 @@
+{pkgs, qemu, disk, overlayPath, safeProductName, updateUrl, ...}:
+let
+   nixos = pkgs.importFromNixos "";
+
+   nextVersion = "9999.99.99-TESTMAGIC";
+
+   # TODO: use a real PlayOS system and also test that MarkGood works
```

Comment:
I withdrew because RAUC (and hence `MarkGood`) cannot work without a GRUB boot (or an emulation of it), which is possible, but requires more work than the wiring you added. Additionally, since e2e tests are supposed to test the "real" system, I think for checking whether `MarkGood` works, a reverse kind of test scenario would be more appropriate: given a "real" bundle, try to update into it from an "older" pseudo-test-system and check that it boots and reaches `MarkGood`. Or don't even bother with the update process and check that the system reaches MarkGood when booted, since it is kinda orthogonal.

As for why the test case fails after the changes: your changes increased the bundle size to more than 600MB, controller limits curl to 10MB, so now the Download phase exceeded the 61s timeout. Added a comment to explain this and bumped the timeout anyway (a7b707f1ea). I think the controller logs with `latest=1.1.1` confused you because they were being dumped in reverse order (bad habit of mine), fixed that in 5d487f3.

I remove this (and many other TODOs) in 30935456b6, since `MarkGood` tests are kind of another territory that needs to be explored separately.

Reply:
Oh wow, yeah, I was totally thrown by the log order! :upside_down_face: 

Things make sense now, thanks.